edx_rest_framework_extensions/permissions.py

```diff
@@ -1,9 +1,71 @@
 """ Permission classes. """
+from oauth2_provider.ext.rest_framework.permissions import TokenHasScope
+from opaque_keys.edx.keys import CourseKey
+from rest_framework.exceptions import PermissionDenied
 from rest_framework.permissions import BasePermission
 
+from edx_rest_framework_extensions.decorators import (
+    jwt_authentication_required,
+    skip_unless_jwt_scopes_enforced,
+)
+from edx_rest_framework_extensions.utils import jwt_decode_handler
+
 
 class IsSuperuser(BasePermission):
     """ Allows access only to superusers. """
 
     def has_permission(self, request, view):
         return request.user and request.user.is_superuser
+
+
+class JwtHasScope(TokenHasScope):
+    """
+    The request is authenticated as a user and the token used has the right scope.
+    """
+
+    @skip_unless_jwt_scopes_enforced
+    @jwt_authentication_required
+    def has_permission(self, request, view):
+        jwt_scopes = jwt_decode_handler(request.auth).get('scopes', [])
+        required_scopes = set(self.get_scopes(request, view))
+        if required_scopes.issubset(jwt_scopes):
+            return True
+
+        if required_scopes:
+            # pylint: disable=attribute-defined-outside-init
+            self.message = {
+                "detail": PermissionDenied.default_detail,
+                "required_scopes": required_scopes,
+            }
+
+        return False
+
+
+class JwtHasContentOrgFilterForRequestedCourse(BasePermission):
+    """
+    The JWT used to authenticate contains the appropriate content provider
+    filter for the requested course resource.
+    """
+
+    @skip_unless_jwt_scopes_enforced
+    @jwt_authentication_required
+    def has_permission(self, request, view):
+        """
+        Ensure that the course_id kwarg provided to the view contains one
+        of the organizations specified in the content provider filters
+        in the JWT used to authenticate.
+        """
+        course_key = CourseKey.from_string(view.kwargs.get('course_id'))
+        jwt_filters = jwt_decode_handler(request.auth).get('filters', [])
+        for jwt_filter in jwt_filters:
+            (provider_type, org_filter) = jwt_filter.split(':')
+            if provider_type == 'content_org' and org_filter == course_key.org:
```

Comment:
Same question as above, should the parsing of `filters` in the JWT token live with the other token parsing code?